azurelinuxagent/common/utils/cryptutil.py

```diff
@@ -92,6 +92,8 @@ def decrypt_p7m(self, p7m_file, trans_prv_file, trans_cert_file, pem_file):
                 "-out", pem_file]
 
             first_proc = subprocess.Popen(first_cmd, stdout=subprocess.PIPE)
+            first_proc.wait()
```

Comment:
Does this have potential to hang indefinitely? Because as we saw `wait()` is uninterruptable

Reply:
We're already using `communicate` a couple of lines down. Is this is a concern, we'd need to fix it there too.

As for this wait, I can replace it with a single `poll` and it works as well; `communicate` already waits for `first_proc` to complete (as well as `second_proc`). This call (or, equally, a `poll` call) just populates the `Popen` object's `returncode` member variable.

Reply:
I would not be too concerned about the wait since those commands complete pretty quickly. The issues we've had is when we wait on processes that can take a long time to complete or never complete.